### PR TITLE
Add more granular exception classes for HTTPError

### LINKED ISSUE #252: Better HTTP Exceptions?
It would be nice to have a little more granular HTTP Exceptions, rather than just a single one.

All HTTPExceptions could inherit from the existing HTTPException class, but we could have ones like:

NotFoundHTTPException(HTTPException):
ForbiddenHTTPException(HTTPException):
GatewayTimeoutHTTPException(HTTPException):

It would be nice for when I'm catching different exceptions and trying to handle each one differently.

### PR DESCRIPTION
**Pull request by bxm156**
Add some common HTTP exception classes that inherit from HTTPError.
Added a common 4xx and 5xx error classes to start with.
fixes #252

### CODE REVIEW

**Comment by macisamuele** [comment]
@bxm156 thanks a lot for your solution ... it's great.
I would ask if you can map all the known standard http responses ... you can find them on the [IANA Standard](http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml).

Maybe you could try to get all of them using this.
```bash
# Classes generation
curl http://www.iana.org/assignments/http-status-codes/http-status-codes-1.csv | \
    grep -vE "(Value,Description,Reference|Unassigned|Unused)" | \
    awk -F, '{ if($1>=400) { gsub(" ","", $2); printf "\n\nclass HTTP%s(HTTPClientError):\n        status = %s\n", $2, $1}}'
```

**Comment by bxm156** [comment]
@macisamuele  sure thing

File: bravado/exception.py [hunk 1]
```diff
@@ -53,10 +53,33 @@ def __str__(self):
...
+
 class HTTPClientError(HTTPError):
     """4xx responses."""
 
 
+class HTTPServerError(HTTPError):
+    """5xx responses."""
+
+
+# The follow are based on the HTTP Status Code Registery at
```
Comment (macisamuele) [inline bravado/exception.py]: nit: Registery -> Registry

Comment (bxm156) [inline bravado/exception.py]: Nooooooo!!!! I was so close to having no spelling errors :(

**Comment by macisamuele** [comment]
@bxm156 Thanks a lot for your contribution.

**Pull request merged**